modules/REST/V2/Languages.php

```diff
@@ -0,0 +1,737 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\REST\V2;
+
+use PLL_Language;
+use PLL_Model;
+use stdClass;
+use WP_Error;
+use WP_REST_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Allows to access languages via the REST API.
+ *
+ * @since 3.7
+ *
+ * @phpstan-type Fields array{
+ *     code?: non-empty-string,
+ *     locale?: non-empty-string,
+ *     name?: non-empty-string,
+ *     direction?: 'ltr'|'rtl',
+ *     order?: int,
+ *     flag?: string,
+ *     set_default_cat?: bool
+ * }
+ */
+class Languages extends WP_REST_Controller {
+	/**
+	 * @var PLL_Model
+	 */
+	public $model;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 *
+	 * @param PLL_Model $model Polylang's model.
+	 */
+	public function __construct( PLL_Model $model ) {
+		$this->namespace = 'pll/v2';
+		$this->rest_base = 'languages';
+		$this->model     = $model;
+	}
+
+	/**
+	 * Registers the routes for languages.
+	 *
+	 * @since 3.7
+	 *
+	 * @return void
+	 */
+	public function register_routes(): void {
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}",
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => '__return_true',
+				),
+				array(
+					'methods'             => WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'create_item' ),
+					'permission_callback' => array( $this, 'create_item_permissions_check' ),
+					'args'                => $this->get_endpoint_args_for_item_schema( WP_REST_Server::CREATABLE ),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			"/{$this->rest_base}/(?P<code>[a-z_-]+)",
```

Comment:
That makes me think of something, does language slug accept figures? If so here is what I suggest:
```suggestion
			"/{$this->rest_base}/(?P<code>[a-zA-Z0-9_-]+)",
```

Reply:
It's the pattern we use for the slug: https://github.com/polylang/polylang/blob/e647cbe80b72416ed00b172c7b0a9cd078cb59c0/admin/admin-model.php#L367.